client/src/app/user/user.service.ts

```diff
@@ -80,6 +80,15 @@ export class UserService {
     return this.observeUserChanges().pipe(take(1));
   }
 
+  getCurrentUserSaved(): Observable<boolean> {
+    return this.observeUserChanges().pipe(
+      take(1),
+      switchMap(user => this.db.userRef(user.id).onceValue()),
+      snapshotToValue,
+      map(user => !!user)
+    );
```

Comment:
I think we can refactor this and use `getUser()` instead. We can refactor `getUser(id)` to make `id` optional. If it doesn't have an `id` it'll use the one attached to the emitted loginUser object from `authService.requireAuthOnce()`.